pkg/core/annotations.go

```diff
@@ -35,3 +37,39 @@ func (c *client) bumpNonceAnnotation(s *v1alpha1.Service) {
 	annotations[buildAnnotation] = strconv.Itoa(i + 1)
 	s.Spec.RunLatest.Configuration.RevisionTemplate.SetAnnotations(annotations)
 }
+
+func (c *client) bumpNonceAnnotationForBuild(s *v1alpha1.Service) {
+	configurationSpec := s.Spec.RunLatest.Configuration
```

Comment:
should we also handle Pinned and Release config modes?

Reply:
not in this PR, maybe separate issue